mmaction/models/backbones/stgcn.py

```diff
@@ -0,0 +1,282 @@
+
+        return x.contiguous(), A
+
+
+@BACKBONES.register_module()
+class STGCN(nn.Module):
+    """Backbone of Spatial temporal graph convolutional networks.
+
+    Args:
+        in_channels (int): Number of channels in the input data.
+        graph_cfg (dict): The arguments for building the graph.
+        edge_importance_weighting (bool): If ``True``, adds a learnable
+            importance weighting to the edges of the graph. Default: True.
+        data_bn (bool): If 'True', adds data normalization to the inputs.
+            Default: True.
+        pretrained (str | None): Name of pretrained model.
+        **kwargs (optional): Other parameters for graph convolution units.
+
+    Shape:
+        - Input: :math:`(N, in_channels, T_{in}, V_{in}, M_{in})`
+        - Output: :math:`(N, num_class)` where
+            :math:`N` is a batch size,
+            :math:`T_{in}` is a length of input sequence,
+            :math:`V_{in}` is the number of graph nodes,
+            :math:`M_{in}` is the number of instance in a frame.
+    """
+
+    def __init__(self,
+                 in_channels,
+                 graph_cfg,
+                 edge_importance_weighting=True,
+                 data_bn=True,
+                 pretrained=None,
+                 **kwargs):
+        super().__init__()
+
+        # load graph
+        self.graph = Graph(**graph_cfg)
+        A = torch.tensor(
+            self.graph.A, dtype=torch.float32, requires_grad=False)
+        self.register_buffer('A', A)
+
+        # build networks
+        spatial_kernel_size = A.size(0)
+        temporal_kernel_size = 9
+        kernel_size = (temporal_kernel_size, spatial_kernel_size)
+        self.data_bn = nn.BatchNorm1d(in_channels *
+                                      A.size(1)) if data_bn else iden
+
+        kwargs0 = {k: v for k, v in kwargs.items() if k != 'dropout'}
+        self.st_gcn_networks = nn.ModuleList((
+            st_gcn_block(
+                in_channels, 64, kernel_size, 1, residual=False, **kwargs0),
+            st_gcn_block(64, 64, kernel_size, 1, **kwargs),
+            st_gcn_block(64, 64, kernel_size, 1, **kwargs),
+            st_gcn_block(64, 64, kernel_size, 1, **kwargs),
+            st_gcn_block(64, 128, kernel_size, 2, **kwargs),
+            st_gcn_block(128, 128, kernel_size, 1, **kwargs),
+            st_gcn_block(128, 128, kernel_size, 1, **kwargs),
+            st_gcn_block(128, 256, kernel_size, 2, **kwargs),
+            st_gcn_block(256, 256, kernel_size, 1, **kwargs),
+            st_gcn_block(256, 256, kernel_size, 1, **kwargs),
+        ))
+
+        # initialize parameters for edge importance weighting
+        if edge_importance_weighting:
+            self.edge_importance = nn.ParameterList([
+                nn.Parameter(torch.ones(self.A.size()))
+                for i in self.st_gcn_networks
+            ])
+        else:
+            self.edge_importance = [1] * len(self.st_gcn_networks)
```

Comment:
not save, they share the same memory. use `[1 for _ in len(self.st_gcn_networks)]`